src/cpp/preprocessor/asm/asm_parser.cpp

```diff
@@ -192,10 +192,10 @@ parse_lines(const std::vector<char>& data, std::string& file)
 
 void
 attach_to_group_directive::
-operate(std::shared_ptr<asm_parser> parserptr, const std::smatch& sm)
+operate(std::shared_ptr<asm_parser> parserptr, const boost::smatch& sm)
 {
   m_parserptr = parserptr;
-  if (sm.size() < 3)
+  if (sm.size() < 3 || !sm[2].matched || sm[2].length() == 0)
```

Comment:
It would be best to refactor this and the following line like with the help of a new inline function:

`verifiy_match(sm, error::error_code::invalid_asm, "Invalid attach_to_group directive argument\n");`


where verify_match is defined as:

```
void verify_match(const boost::smatch& sm, error::error_code ecode, const char *msg){
    if (sm.size() < 3 || !sm[2].matched || sm[2].length() == 0)
        throw error(ecode, msg);
}
```